src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5339,23 +5445,23 @@ static synchronized long getColumnEncryptionKeyCacheTtl() {
      * @param directSql
      *      Whether the statement handle is direct SQL (true) or a cursor (false)
      */
-    final void enqueuePreparedStatementDiscardItem(int handle, boolean directSql) {
-        if (this.getConnectionLogger().isLoggable(java.util.logging.Level.FINER))
-            this.getConnectionLogger().finer(this + ": Adding PreparedHandle to queue for un-prepare:" + handle);
+    final void enqueueUnprepareStatementHandle(PreparedStatementHandle statementHandle) {
+        if (loggerExternal.isLoggable(java.util.logging.Level.FINER))
+            loggerExternal.finer(this + ": Adding PreparedHandle to queue for un-prepare:" + statementHandle.getHandle());
 
         // Add the new handle to the discarding queue and find out current # enqueued.
-        this.discardedPreparedStatementHandles.add(new PreparedStatementDiscardItem(handle, directSql));
-        this.discardedPreparedStatementHandleQueueCount.incrementAndGet();
+        discardedPreparedStatementHandles.add(statementHandle);
+        if (0 < discardedPreparedStatementHandleCount.incrementAndGet())
+            handlePreparedStatementDiscardActions(false);
     }
 
-
     /**
      * Returns the number of currently outstanding prepared statement un-prepare actions.
      * 
      * @return Returns the current value per the description.
      */
-    public int getDiscardedServerPreparedStatementCount() {
```

Comment:
We debated the name of these methods before the last dev release, not sure it makes sense to change this public API now.